webapi/CopilotChat/Skills/ChatSkills/ExternalInformationSkill.cs

```diff
@@ -88,12 +88,7 @@ public class ExternalInformationSkill
             string planJson = JsonSerializer.Serialize(deserializedPlan.Plan);
             // Reload the plan with the planner's kernel so
             // it has full context to be executed
-            var newPlanContext = new SKContext(
-                null,
-                this._planner.Kernel.Memory,
-                this._planner.Kernel.Skills,
-                this._planner.Kernel.Log
-            );
+            var newPlanContext = new SKContext(null, this._planner.Kernel.Skills, this._planner.Kernel.Log);
```

Comment:
Does the planner require memory? If so, does the kernel for the planner point to the same memory store as the other kernel? @teresaqhoang

Reply:
If it does, it must be using the memory from kernel.Memory...

Reply:
Yeah, but the planner uses a kernel that is not the one used by chat.

Reply:
If it's obtained through dependency injection, it should be the same singleton instance

Reply:
So I just double-checked. The ActionPlanner doesn't use memory. The sequential one does to generate and store embeddings for the function descriptions.

Also, you were right, the planner kernel wasn't assigned the memory from dependency injection.

I'm making the necessary changes so that we are goo when dynamically adding functions to the kernel.

Reply:
The Planner doesn't use memory, but we're planning to default to SequentialPlanner, and there's talk of using the planner to load memory as needed, as opposed to in each chat request, so I would recommend at least putting a TODO in here to pass memory into planner. Check with Ben on what issue would be best to tag.

Reply:
OK, then.  I'll just give the planner access to memory through dependency injection (one line changed) but won't change the current planner behavior. If and when we do, memory will be there.